File: LeetCode/1414.和为K的最少斐波那契数字数目/1414.和为K的最少斐波那契数字数目.cpp
class Solution {
public:
    void recFibonacci(int &k, int n1, int n2, int &count){
        int nFib = n1 + n2;
        if (k - nFib < 0){
            return;
        }else{
             recFibonacci(k, n2, nFib, count);
        }
        if (k - nFib >= 0){
            k = k - nFib;
            count += 1;
        }
        return;
    }
    int findMinFibonacciNumbers(int k) {
        int count = 0;
        recFibonacci(k, 1, 1, count);
        if(k == 1){
            count += 1;
        }
        return count;
    }
};

File: LeetCode/1414.和为 K 的最少斐波那契数字数目/1414.和为K的最少斐波那契数字数目.go
func findMinFibonacciNumbers(k int) (ans int) {
    f := []int{1, 1}
    for f[len(f) - 1] < k {
        f = append(f, f[len(f) - 1] + f[len(f) - 2])
    } 
    for i:= len(f) -1; i >= 0 && k >= 0; i-- {
        if k - f[i] >= 0 {
            k = k -f[i]
            ans += 1
        }
    }
    return ans
}